src/main/java/gift/entity/product/Product.java

```diff
@@ -0,0 +1,116 @@
+package gift.entity.product;
+
+import gift.dto.ProductResponse;
+import gift.entity.product.value.ProductId;
+import gift.entity.product.value.ProductImageUrl;
+import gift.entity.product.value.ProductName;
+import gift.entity.product.value.ProductPrice;
+import java.util.Objects;
+
+public class Product {
+
+    private final ProductId id;
+    private final ProductName name;
+    private final ProductPrice price;
+    private final ProductImageUrl imageUrl;
+    private final boolean hidden;
+
+    private Product(ProductId id, ProductName name, ProductPrice price, ProductImageUrl imageUrl,
+            boolean hidden) {
+        this.id = id;
+        this.name = name;
+        this.price = price;
+        this.imageUrl = imageUrl;
+        this.hidden = hidden;
+    }
+
+    public static Product of(Long id, String name, int price, String imageUrl, boolean hidden) {
+        return new Product(
+                new ProductId(id),
+                new ProductName(name),
+                new ProductPrice(price),
+                new ProductImageUrl(imageUrl),
+                hidden
+        );
+    }
+
+    public static Product of(Long id, String name, int price, String imageUrl) {
+        return new Product(
+                new ProductId(id),
+                new ProductName(name),
+                new ProductPrice(price),
+                new ProductImageUrl(imageUrl),
+                false
+        );
+    }
+
+    public static Product of(String name, int price, String imageUrl) {
+        return new Product(
+                null,
+                new ProductName(name),
+                new ProductPrice(price),
+                new ProductImageUrl(imageUrl),
+                false
+        );
+    }
```

Comment:
```suggestion
    public static Product of(Long id, String name, int price, String imageUrl, boolean hidden) {
        return new Product(
                new ProductId(id),
                new ProductName(name),
                new ProductPrice(price),
                new ProductImageUrl(imageUrl),
                hidden
        );
    }

    public static Product of(Long id, String name, int price, String imageUrl) {
        return of(id, name, price, imageUrl, false);
    }

    public static Product of(String name, int price, String imageUrl) {
        return of(null, name, price, imageUrl, false);
    }
```
- 중복코드는 습관적으로 제거해주세요.